specs/ov_management_api.md

```diff
@@ -0,0 +1,171 @@
+The method by which clients are provided with credentials is outside the scope of this specification.
+
+#### X.509 Client Certificate
+
+In this authentication method, a client will send its certificate as part of the TLS connection handshake.
+For this, the certificate needs to be signed by a Certificate Authority that the server trusts.
+The `username` in the protocol is the value of the `CN` field in the `Subject` of the certificate.
+
+#### OAuth2 Bearer token in Authorization header
+
+In this authentication method, the client will send an OAuth2 Bearer token, as per [RFC6750, section 2.1](https://datatracker.ietf.org/doc/html/rfc6750#section-2.1).
+How the server determines a `username` for this authentication method is outside the scope of this specification.
+
+### Responses
+
+Requests that got parsed and executed succesfully will have a Status code in the Succesful range (section 6.3, RFC 7231).
+If an error occured during processing, a Status code in either the Client Error (section 6.5, RFC 7231) or Server Error (section 6.6, RFC 7231) will be returned.
+
+Responses will have `Content-Type: application/json` and consist of JSON strings.
+
+#### Error responses
+
+Error responses will consist of JSON objects, with at least the following keys:
+
+- `error_code`: An operation-specific string error code.
+- `error_details`: A JSON object with keys defined by the specific `error_code` value.
+
+## Ownership Voucher upload
+
+HTTP Request context: `POST $base/v1/ownership_voucher`.
+
+This endpoint can be used to upload a batch of new ownership vouchers.
+A header is sent with the number of vouchers to be uploaded, so the Owner Onboarding Service can verify that it did in fact receive (and process) every ownership voucher.
+This endpoint will accept either raw ownership vouchers (CBOR encoding), or PEM wrapper ownership vouchers, with the `OWNERSHIP VOUCHER` type, which one will be parsed/expected is determined by the `Content-Type` request header: `application/cbor` for CBOR, and `application/x-pem-file` for PEM.
+In the case of CBOR encoded vouchers, they should just be appended to each other as a byte stream.
+In the case of PEM encoded vouchers, vouchers should be appended to each other, adding a newline between the different entries.
+The request SHOULD contain a header `X-Number-Of-Vouchers`, containing the number of Ownership Vouchers being uploaded.
+If this number diverges from the number of vouchers the server parsed, they should refuse the entire request.
+
+A successful response will contain a JSON list containing objects, which each have at least the following keys:
+
+- `guid`: the FDO GUID of the Ownership Voucher.
+
+### Error codes
+
+- `incomplete_voucher`: when an uploaded voucher was incomplete, `error_details` contains at least the key `parsed_correctly`, containing the number of Ownership Vouchers succesfully parsed.
```

Comment:
The rest of error codes follow the structure <name>: <reason><DOT><what may `error_details` contain>. In this case, instead of the dot, we're using a comma, which I guess is not intentional.